allennlp/nn/initializers.py

```diff
@@ -193,9 +214,22 @@ def __init__(self, initializers: List[Tuple[str, Initializer]] = None) -> None:
         initializers : ``List[Tuple[str, Initializer]]``, optional (default = [])
             A list mapping parameter regexes to initializers.  We will check each parameter against
             each regex in turn, and apply the initializer paired with the first matching regex, if
-            any.
+            any. If "prevent" is assigned to any regex, then it will override and prevent the matched
+            parameters to be initialzed.
         """
-        self._initializers = initializers or []
+
+        self._initializers: List[Tuple[str, Initializer]] = []
+        initializers = initializers or []
+        self._prevent_regex = None
+        prevent_regexes = []
+        for initializer in initializers:
+            if isinstance(initializer[1], PreventRegexInitializer):
+                prevent_regexes.append(initializer[0])
```

Comment:
To avoid having to type check for this particular initialiser, I think it would be better to add another argument to the `InitializerApplicator`: `prevent_regexes: List[str] = None`. 

This way, you can just pass a list of regexes for parameters to be ignored. To instanciate this class we just need to modify `from_params` to pop the "prevent_regexes" key before building the other initialisers You can also delete the `PreventRegexInitializer` class in this case. Do you see any problems with that?

Reply:
Yes I agree that `PreventRegexInitializer` seems hacky solution and its better to delete it. Having `prevent_regexes: List[str] = None` argument in `InitializerApplicator` and removing `PreventRegexInitializer` seems much cleaner. I will make these changes. Thank you for the suggestions.

However, poping `"prevent_regexes"` key in `from_params` of `InitializerApplicator` is not possible since `params` argument is a List of tuples `List[Tuple[str, Params]]` as following:

```
[
    ["parameter_regex_match1",
        {"type": "normal","mean": 0.01, "std": 0.1}
    ],
    ["parameter_regex_match2", "uniform"]
]
```
How about the following: To add prevent regex we add `(".*_my_prevention_regex_*",  "prevent")` tuples in the above list. And in `from_params`, from `params` list, I extract and remove the tuples with second item "prevent" and make `prevent_regexes` out of first items. And finally initialize `Initializer`s:
```python
def from_params(cls, params: List[Tuple[str, Params]]) -> "InitializerApplicator":
    prevent_regexes = [ param[0] for param in params if param[1] is "prevent"]
    params = [ param for param in params if param[1] is not "prevent"]

    initializers = [(name, Initializer.from_params(init_params, prevent_regexes)) for name, init_params in params]
    return InitializerApplicator(initializers)
```
How does this seem? Please correct me if I misinterpreted your suggestion.

Reply:
Ah yes, my bad - go ahead, that seems equally as good.

Reply:
Cool, Thanks!